分词器初始化完成。词汇表大小: 10004
加载并拼接完成，文本长度：108021386字符
开始对全部文本进行编码...
文本编码完成。总 Token 数量: 9999999
已创建 9999899 个训练样本，每个长度为 100
真实数据 DataLoader 已准备好
模型已创建并移动到 cuda
模型总参数量: 57,904,916

--- 开始使用真实数据进行训练 ---
Training:   0%|                                                                                                                                                                                                                                                              | 0/281248 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 229, in <module>
    train_process(config)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 198, in train_process
    trainer.train(num_epochs=config['num_epochs'], train_loader=train_loader, val_loader=val_loader)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 83, in train
    train_loss = self._train_epoch(train_loader)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 51, in _train_epoch
    outputs = self.model(inputs)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\littleTransformer.py", line 381, in forward
    x = self.token_embedding(src) * math.sqrt(self.d_model)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'DecoderOnlyTransformer' object has no attribute 'd_model'
Traceback (most recent call last):
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 229, in <module>
    train_process(config)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 198, in train_process
    trainer.train(num_epochs=config['num_epochs'], train_loader=train_loader, val_loader=val_loader)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 83, in train
    train_loss = self._train_epoch(train_loader)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\train.py", line 51, in _train_epoch
    outputs = self.model(inputs)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\Ruizhe\Desktop\Study\Code\AI\transformer\model\littleTransformer.py", line 381, in forward
    x = self.token_embedding(src) * math.sqrt(self.d_model)
  File "E:\Software\Python_env\Anaconda3\envs\hf_LLM_env\lib\site-packages\torch\nn\modules\module.py", line 1695, in __getattr__
    raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'DecoderOnlyTransformer' object has no attribute 'd_model'
